packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -88,11 +89,15 @@ const getLocationSearch = (req) => {
         const values = Array.isArray(value) ? value : [value]
 
         values.forEach((v) => {
-            newParams.append(key, v)
+            // Because URLSearchParams constructor perceives + as space, but append does not
+            // we want to make sure we run the key-value through new URLSearchParams constructor to parse + as space
+            // before we append it to newParams
+            const temp = new URLSearchParams(`${key}=${v}`)
```

Comment:
While URLSearchParams construct treats plus sign (+) as space when you first initialized, its append method does not, and we want to retain that behavior during appending by runing the key-val through URLSearchParams, before passing them into the append method